### PR TITLE
added half-even rounding

### PR DESCRIPTION
**Pull request by aywan**
Hi!
I add half-even rounding, also known as banker's rounding.
More at wiki https://en.wikipedia.org/wiki/Rounding#Round_half_to_even

### CODE REVIEW

**Comment by peter279k** [comment]
Hi @aywan, thanks for your PR.

And it looks like a `IEEE-754` standard to calculate the decimal values.

File: tests/Unit/BCTest.php [hunk 1]
```diff
@@ -284,6 +284,47 @@ public function shouldRound($expected, $number, $precision = 0): void
...
+            ['0.08', '0.075', 2],
+            ['0.08', '0.085', 2],
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider roundHalfEvenProvider
+     * @param string $expected
+     * @param int|float|string $number
```
Comment (peter279k) [inline tests/Unit/BCTest.php]: According to the test cases from this data provider, the `$number` `@param` comment annotation should be always `string`.

File: tests/Unit/BCTest.php [hunk 2]
```diff
@@ -284,6 +284,47 @@ public function shouldRound($expected, $number, $precision = 0): void
...
+     * @param int $precision
+     */
+    public function shouldRoundHalfEven($expected, $number, $precision = 0): void
+    {
+        self::assertSame($expected, BC::roundHalfEven((string)$number, $precision));
```
Comment (peter279k) [inline tests/Unit/BCTest.php]: The `$number` always is the `string` and we don't need to add `(string)` to cast this variable.

**Comment by aywan** [comment]
Hi @peter279k 
Thank you for review.
Yes, this one of rounding using at IEEE-754.

File: tests/Unit/BCTest.php (hunk 2 above)
Comment (aywan) [reply tests/Unit/BCTest.php]: fixed

File: tests/Unit/BCTest.php (hunk 1 above)
Comment (aywan) [reply tests/Unit/BCTest.php]: fixed

File: tests/Unit/BCTest.php [hunk 3]
```diff
@@ -284,6 +284,47 @@ public function shouldRound($expected, $number, $precision = 0): void
...
+            ['2.4', '2.35', 1],
+            ['0', '0.005', 2],
+            ['0.02', '0.015', 2],
+            ['0.02', '0.025', 2],
+            ['0.04', '0.035', 2],
+            ['0.04', '0.045', 2],
+            ['0.06', '0.055', 2],
+            ['0.06', '0.065', 2],
+            ['0.08', '0.075', 2],
+            ['0.08', '0.085', 2],
```
Comment (krowinski) [inline tests/Unit/BCTest.php:313]: Hi @aywan can you add bigger numbers for example:

`
            ['1234567890', '9876543210', '12193263111263526900', 2],
            ['2.5', '1.5', '3.75', 2],
            ['2.555', '1.555', '3.97', 2],
            [9.9999E-2, -5.6E-2, '-0.005599944', 9], 
`

you can use wolframalpha.com to confirm them 

tx

Comment (aywan) [reply tests/Unit/BCTest.php:313]: Hi @krowinski 
Sorry, lost your comment. Can add it if necessary.

Comment (krowinski) [reply tests/Unit/BCTest.php:313]: yes pls

**Pull request merged**